Enforce unique workflow names

ivov opened the pull request:
This PR
- adds migrations to enforce unique workflow names, and
- creates a `/workflows/new` endpoint for workflow name defaults and offsets.

**Important**: For pre-existing duplicate workflow names, this renames the workflow by appending an incrementing number.

ivov committed:
Set default dates in entities

ivov committed:
:shirt: Fix lint in migrations

ivov committed:
:zap: Implement unique workflow names

ivov committed:
:zap: Create /workflows/new endpoint

ivov committed:
:zap: Add query string params to /workflows/new

ivov committed:
:construction: Adjust to new qs requirements

ivov committed:
:zap: Create migrations for unique workflow names

ivov committed:
:zap: Adjust quotes in MySQL migration

ivov committed:
:zap: Adjust quotes in Postgres migration

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=1784) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=1784) before we can accept your contribution.<br/>**8** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: RicardoE105<br/>:white_check_mark: ivov<br/>:white_check_mark: ahsanv<br/>:white_check_mark: janober<br/>:white_check_mark: krynble<br/>:white_check_mark: MedAliMarz<br/>:white_check_mark: DeskYT<br/>:white_check_mark: harshil1712<br/>:x: snyk-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=1784) it.</sub>

ivov committed:
:zap: Refactor endpoint per new guidelines

ivov committed:
:zap: Refactor to support numeric suffixes

ivov committed:
Merge branch 'master' into unique-workflow-names

ivov committed:
Merge branch 'master' into set-default-dates

ivov committed:
Merge branch 'set-default-dates' into unique-workflow-names

ivov committed:
:shirt: Lint migrations for unique workflow names

ivov committed:
:zap: Add migrations set default dates to indexes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=1784) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=1784) before we can accept your contribution.<br/>**12** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: ivov<br/>:white_check_mark: ahsanv<br/>:white_check_mark: krynble<br/>:white_check_mark: janober<br/>:white_check_mark: MedAliMarz<br/>:white_check_mark: harshil1712<br/>:white_check_mark: mdasmendel<br/>:white_check_mark: DeskYT<br/>:white_check_mark: RicardoE105<br/>:white_check_mark: DerEnderKeks<br/>:white_check_mark: tanay1337<br/>:white_check_mark: 1izardo<br/>:x: snyk-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=1784) it.</sub>

ivov committed:
Merge branch 'add-tags' into unique-workflow-names

ivov committed:
:zap: Lowercase default workflow name

ivov committed:
:zap: Add prefixes to set default dates migration

ivov committed:
:zap: Fix indentation on default dates migrations

ivov committed:
:zap: Add temp ts-ignore for unrelated change

ivov committed:
:zap: Adjust default dates migration for MySQL

Remove change to data column in credentials_entity, already covered by Omar's migration. Also, fix quotes from table prefix addition.

ivov committed:
:zap: Adjust quotes in dates migration for PG

ivov committed:
:hammer: Remove mistakenly added nullable property

ivov committed:
:fire: Remove unneeded index drop-create (PG)

ivov committed:
:fire: Remove unneeded table copying

ivov committed:
:zap: Merge dates migration with tags migration

ivov committed:
:hammer: Refactor endpoint and make wf name env

ivov committed:
:zap: Update name missing toast message

ivov committed:
:zap: Update workflow already exists message

ivov committed:
Merge branch 'master' into unique-workflow-names

ivov committed:
Merge branch 'add-tags' of https://github.com/n8n-io/n8n into add-tags

ivov committed:
Merge branch 'add-tags' into unique-workflow-names

BHesseldieck reviewed (approved):
🤘🏼 🎉
